pkg/config/config.go

```diff
@@ -81,14 +84,9 @@ func (cfg *Config) BindFlags() {
 		"Configures the logger to use a Zap development config (encoder=consoleEncoder,logLevel=Debug,stackTraceLevel=Warn, no sampling), "+
 			"otherwise a Zap production config will be used (encoder=jsonEncoder,logLevel=Info,stackTraceLevel=Error), sampling).",
 	)
-	flag.StringVar(
-		&cfg.AccountID, flagAWSAccountID,
-		"",
-		"The AWS Account ID in which the service controller will create resources",
-	)
```

Comment:
@RedbackThomson @vijtrip2 we use the `--aws-account-id` flag when setting up `make kind-test` using kustomize and non-IRSA setups. Is there a reason to remove this?

Reply:
@jaypipes, in those tests we inject the temp aws creds(AWS_ACCESS_KEY_ID , ...)
And if we keep doing that, this aws-account-id will be correctly determined(using sts call) without needing to pass it as flag value

As pointed out by Nick, this flag can cause inconsistencies and confusion between actual credentials being used vs account-id printed in logs

Reply:
>"we use the --aws-account-id flag when setting up make kind-test using kustomize and non-IRSA setups"

I did add TODO's in the PR description to remove this flag from the test-infra repo

Reply:
I've been thinking about this and yeah I don't see a reason to need an `aws-account-id` flag. The only place where we use this account ID is when we add the "OwnerAccountID" field to an object. That is, if we haven't assumed another role as part of the configmap annotation. So it's safe to say that whatever credentials we use with `sts get-caller-identity` will be correct.

Reply:
`aws-account-id` is used in CARM to keep trace of which resource is create in which account (mainly in the logs of resource reconciliation)

Reply:
> this will definitely impact the way we CARM config/patterns in logs/configMaps/Annotations

Can you elaborate on this, @A-Hilaly ? 
All this PR is doing is just removing option for customers and automatically detect account-id. aws-account-id will still be part of Config and there are no changes in any types. I wonder how it will impact any functionalities for CARM.

Reply:
CARM assumes the `--aws-account-id` as the default AWS Account ID for any resources that do not have overrides in the namespace or on the resource annotation. This pull request will infer `--aws-account-id` from the credentials, rather than rely on it being hardcoded as an argument. All of the CARM logic will remain the same, and any other reconciliation logic will too.

I think it would be a great addition (not for this PR) to have a log statement when the controller starts that prints the default account ID and role that is being used by the controller. That is, if there are no overrides, what role is creating resources. [@komish](https://github.com/komish) pointed out that this is not clear at the moment, and makes it difficult to debug when you are trying to use CARM.